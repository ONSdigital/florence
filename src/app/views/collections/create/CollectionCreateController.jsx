import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import collections from "../../../utilities/api-clients/collections";
import teams from "../../../utilities/api-clients/teams";
import notifications from "../../../utilities/notifications";
import log from "../../../utilities/logging/log";
import CollectionCreate from "./CollectionCreate";
import { createCollectionRequest } from "../../../config/thunks";
import { updateAllTeamIDsAndNames, updateAllTeams } from "../../../config/actions";
import collectionValidation from "../validation/collectionValidation";
import { UNIQ_NAME_ERROR } from "../../../constants/Errors";
import { getActive, getCollectionCreating } from "../../../config/selectors";

const propTypes = {
    user: PropTypes.shape({
        userType: PropTypes.string.isRequired,
    }).isRequired,
    allTeams: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired,
        })
    ),
    dispatch: PropTypes.func.isRequired,
    collections: PropTypes.array,
};

export class CollectionCreateController extends Component {
    constructor(props) {
        super(props);

        this.state = {
            newCollectionDetails: {
                name: {
                    value: "",
                    errorMsg: "",
                },
                type: "scheduled",
                publishDate: {
                    value: "",
                    errorMsg: "",
                },
                publishTime: {
                    value: "09:30",
                    errorMsg: "",
                },
                pendingDeletes: [],
                teams: [],
                release: {
                    date: "",
                    title: "",
                    uri: "",
                    isProvisional: null,
                    errorMsg: "",
                },
                scheduleType: "custom-schedule",
            },
            isGettingTeams: true,
            isSubmitting: false,
            showScheduleByRelease: false,
            updatedAllTeams: null,
        };

        this.blankNewCollectionDetails = this.state.newCollectionDetails;
    }

    UNSAFE_componentWillMount() {
        this.getAllTeams();
    }

    getAllTeams() {
        teams
            .getAll()
            .then(teams => {
                if (teams.length === 0) {
                    this.setState({ isGettingTeams: false });
                    return;
                }
                const allTeams = teams.map(team => {
                    return { id: team.id.toString(), name: team.name };
                });
                this.setState({
                    isGettingTeams: false,
                });
                this.props.dispatch(updateAllTeamIDsAndNames(allTeams));

                // Not needed for this screen but keeps teams array up-to-date for the teams screen
                this.props.dispatch(updateAllTeams(teams));
            })
            .catch(error => {
                switch (error.status) {
                    case 401: {
                        // This is handled by the request function, so do nothing here
                        break;
                    }
                    case "RESPONSE_ERR": {
                        const notification = {
                            type: "warning",
                            message: "An error's occurred whilst trying to get teams. You can still create the collection without teams, or refresh",
                            isDismissable: true,
                        };
                        notifications.add(notification);
                        break;
                    }
                    case "UNEXPECTED_ERR": {
                        const notification = {
                            type: "warning",
                            message:
                                "An unexpected error's occurred whilst trying to get teams. You can still create the collection without teams, or refresh",
                            isDismissable: true,
                        };
                        notifications.add(notification);
                        break;
                    }
                    case "FETCH_ERR": {
                        const notification = {
                            type: "warning",
                            message: "There's been a network error whilst trying to get teams. Try refresh the page",
                            isDismissable: true,
                        };
                        notifications.add(notification);
                        break;
                    }
                }

                console.error("Error fetching all teams:\n", error);
            });
    }

    handleCollectionNameChange = event => {
        const collectionName = {
            value: event.target.value,
            errorMsg: "",
        };

        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            name: collectionName,
        };
        this.setState({ newCollectionDetails });
    };

    handleTeamSelection = event => {
        if (!event.target.value) {
            return;
        }

        const teamID = event.target.value;

        // check if team already exists in collection
        for (let i = 0; i < this.state.newCollectionDetails.teams.length; i++) {
            if (this.state.newCollectionDetails.teams[i].id === teamID) {
                return;
            }
        }

        // get info for selected team from teams list in state
        const selectedTeam = this.props.allTeams.find(team => {
            return team.id === teamID;
        });
        const allTeams = this.state.updatedAllTeams || this.props.allTeams || [];
        const updatedAllTeams = allTeams.map(team => {
            if (team.id == teamID) {
                return {
                    ...team,
                    disabled: true,
                };
            }
            return team;
        });

        const teams = [...this.state.newCollectionDetails.teams, selectedTeam];

        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            teams: teams,
        };
        this.setState({
            newCollectionDetails,
            updatedAllTeams,
        });
    };

    handleRemoveTeam = teamToRemove => {
        const updatedTeams = this.state.newCollectionDetails.teams.filter(team => {
            return team.id !== teamToRemove.id;
        });
        const updatedAllTeams = this.state.updatedAllTeams.map(team => {
            if (team.id === teamToRemove.id) {
                return {
                    ...team,
                    disabled: false,
                };
            }
            return team;
        });

        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            teams: updatedTeams,
        };
        this.setState({
            newCollectionDetails,
            updatedAllTeams,
        });
    };

    handleCollectionTypeChange = event => {
        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            type: event.value,
            release: {
                uri: "",
                date: "",
                title: "",
                errorMsg: "",
            },
        };
        this.setState({ newCollectionDetails });
    };

    handleScheduleTypeChange = event => {
        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            scheduleType: event.value,
        };
        this.setState({ newCollectionDetails });
    };

    handlePublishDateChange = event => {
        const publishDate = {
            value: event.target.value,
            errorMsg: "",
        };

        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            publishDate: publishDate,
        };
        this.setState({ newCollectionDetails });
    };

    handlePublishTimeChange = event => {
        const publishTime = {
            value: event.target.value,
            errorMsg: "",
        };

        const newCollectionDetails = {
            ...this.state.newCollectionDetails,
            publishTime: publishTime,
        };
        this.setState({ newCollectionDetails });
    };

    handleAddRelease = event => {
        event.preventDefault();
        this.setState({ showScheduleByRelease: true });
    };

    handleSelectRelease = release => {
        this.setState(state => ({
            newCollectionDetails: {
                ...state.newCollectionDetails,
                release: {
                    uri: release.uri,
                    title: release.title,
                    date: release.releaseDate,
                    isProvisional: release.isProvisional,
                    errorMsg: "",
                },
            },
            showScheduleByRelease: false,
        }));
    };

    handleCloseRelease = () => {
        this.setState({ showScheduleByRelease: false });
    };

    makePublishDate() {
        if (this.state.newCollectionDetails.type !== "scheduled") {
            return null;
        }

        if (this.state.newCollectionDetails.scheduleType === "calender-entry-schedule") {
            return this.state.releaseDateISO;
        }

        const date = this.state.newCollectionDetails.publishDate.value;
        const time = this.state.newCollectionDetails.publishTime.value;
        return new Date(date + " " + time).toISOString();
    }

    mapStateToPostBody() {
        try {
            return {
                name: this.state.newCollectionDetails.name.value,
                type: this.state.newCollectionDetails.type,
                publishDate: this.makePublishDate(),
                teams: this.state.newCollectionDetails.teams.map(team => {
                    return team.name;
                }),
                collectionOwner: this.props.user.userType,
                releaseUri:
                    this.state.newCollectionDetails.scheduleType === "calender-entry-schedule" ? this.state.newCollectionDetails.release.uri : null,
            };
        } catch (error) {
            log.event("Error mapping new collection state to POST body", log.error(error));
            console.error("Error mapping new collection state to POST body" + error);
        }
    }

    handleSubmit = event => {
        event.preventDefault();
        this.setState({ isSubmitting: true });

        let hasError = false;
        let newCollectionDetails = this.state.newCollectionDetails;
        const validatedName = collectionValidation.name(this.state.newCollectionDetails.name.value, this.props.collections);

        if (!validatedName.isValid) {
            const collectionName = {
                value: this.state.newCollectionDetails.name.value,
                errorMsg: validatedName.errorMsg,
            };

            newCollectionDetails = {
                ...newCollectionDetails,
                name: collectionName,
            };
            hasError = true;
        }

        if (this.state.newCollectionDetails.type === "scheduled" && this.state.newCollectionDetails.scheduleType === "custom-schedule") {
            const validatedDate = collectionValidation.date(this.state.newCollectionDetails.publishDate.value);
            if (!validatedDate.isValid) {
                const collectionDate = {
                    value: this.state.newCollectionDetails.publishDate.value,
                    errorMsg: validatedDate.errorMsg,
                };

                newCollectionDetails = {
                    ...newCollectionDetails,
                    publishDate: collectionDate,
                };
                hasError = true;
            }
        }

        if (this.state.newCollectionDetails.type === "scheduled" && this.state.newCollectionDetails.scheduleType === "custom-schedule") {
            const validatedTime = collectionValidation.time(this.state.newCollectionDetails.publishTime.value);
            if (!validatedTime.isValid) {
                const collectionTime = {
                    value: this.state.newCollectionDetails.publishTime.value,
                    errorMsg: validatedTime.errorMsg,
                };

                newCollectionDetails = {
                    ...newCollectionDetails,
                    publishTime: collectionTime,
                };
                this.setState({
                    newCollectionDetails: newCollectionDetails,
                    isSubmitting: false,
                });
                hasError = true;
            }
        }

        if (this.state.newCollectionDetails.type === "scheduled" && this.state.newCollectionDetails.scheduleType === "calender-entry-schedule") {
            const release = this.state.newCollectionDetails.release;
            const validatedRelease = collectionValidation.release(release);
            if (!validatedRelease.isValid) {
                const collectionRelease = {
                    ...release,
                    errorMsg: validatedRelease.errorMsg,
                };

                newCollectionDetails = {
                    ...newCollectionDetails,
                    release: collectionRelease,
                };
                this.setState({
                    newCollectionDetails,
                    isSubmitting: false,
                });
                hasError = true;
            }
        }

        if (hasError) {
            this.setState({
                newCollectionDetails,
                isSubmitting: false,
            });
            return;
        }
        this.props.dispatch(createCollectionRequest(this.mapStateToPostBody()));

        this.setState({
            isSubmitting: false,
        });
    };

    render() {
        return (
            <>
                <CollectionCreate
                    newCollectionDetails={this.state.newCollectionDetails}
                    handleCollectionNameChange={this.handleCollectionNameChange}
                    handleTeamSelection={this.handleTeamSelection}
                    handleRemoveTeam={this.handleRemoveTeam}
                    handleCollectionTypeChange={this.handleCollectionTypeChange}
                    showScheduleOptions={this.state.newCollectionDetails.type === "scheduled"}
                    showCustomScheduleOptions={this.state.newCollectionDetails.scheduleType === "custom-schedule"}
                    handleSelectRelease={this.handleSelectRelease}
                    handleCloseRelease={this.handleCloseRelease}
                    hasTeams={!this.state.isGettingTeams}
                    allTeams={this.state.updatedAllTeams || this.props.allTeams || []}
                    handleScheduleTypeChange={this.handleScheduleTypeChange}
                    handlePublishTimeChange={this.handlePublishTimeChange}
                    handlePublishDateChange={this.handlePublishDateChange}
                    handleAddRelease={this.handleAddRelease}
                    showScheduleByRelease={this.state.showScheduleByRelease}
                    onSubmit={this.handleSubmit}
                    isSubmitting={this.state.isSubmitting}
                />
            </>
        );
    }
}

CollectionCreateController.propTypes = propTypes;

function mapStateToProps(state) {
    return {
        allTeams: state.state.teams.allIDsAndNames,
        newCollection: getActive(state.state),
    };
}

export default connect(mapStateToProps)(CollectionCreateController);
