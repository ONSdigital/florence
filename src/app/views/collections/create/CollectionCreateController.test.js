import React from 'react';
import { CollectionCreateController } from './CollectionCreateController';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

jest.mock('../../../utilities/notifications', () => {
    return {
        add: jest.fn(() => {
            //
        })
    }
});

jest.mock('../../../utilities/date', () => {
    return {
        format: jest.fn(() => {
            return "2017-10-12" // the adding year, formatting and getNow functionality is all tested in the date class so we can just hardcode this response
        }),
        getNow: jest.fn(() => {
            //
        }),
        addYear: jest.fn(() => {
            //
        })
    }
});

jest.mock('../../../utilities/log', () => {
    return {
        add: function() {
            //
        },
        eventTypes: {}
    }
});

jest.mock('../../../utilities/api-clients/teams', () => {
    return {
        getAll: jest.fn(() => {
            return Promise.resolve([{id: 1, name: "Team 1"}, {id: 2, name: "Team 2"}])
        })
    }
});

const newCollectionDetails = {
    name: {
        value: "",
        errorMsg: ""
    },
    type: "scheduled",
    publishDate: {
        value: "",
        errorMsg: ""
    },
    publishTime: {
        value: "09:30",
        errorMsg: ""
    },
    scheduleType: "custom-schedule",
    pendingDeletes: [],
    teams: [],
    collectionOwner: "ADMIN",
    release: {
        uri: "",
        date: "",
        title: "",
        errorMsg: ""
    }
};

const expectedPostBody = {
    collectionOwner: "ADMIN",
    name: "Test collection",
    publishDate: "2017-12-07T09:30:00.000Z",
    releaseUri: null,
    teams: ["Team 1"],
    type: "scheduled"
}

const mockedTeams = [{id: "1", name: "Team 1"}, {id: "2", name: "Team 2"}];

const defaultProps = {
    user: {
        userType: "ADMIN"
    },
    onSuccess: function(){},
    dispatch: () => {},
    allTeams: []
};

test("Create collection form matches stored snapshot", () => {
    const component = renderer.create(
        <CollectionCreateController {...defaultProps} />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test("Handle collection name change updates state correctly", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    expect(component.update().state().newCollectionDetails.name.value).toBe("");
    component.instance().handleCollectionNameChange({target: {value: "Test collection"}});
    expect(component.update().state().newCollectionDetails.name.value).toBe("Test collection");
});

test("Handle team selection change updates state correctly", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    component.setProps({ allTeams: mockedTeams });
    expect(component.update().state().newCollectionDetails.teams).toHaveLength(0);
    component.instance().handleTeamSelection({target: {value: "1"}});
    expect(component.update().state().newCollectionDetails.teams).toContainEqual({id: "1", name: "Team 1"});
});

describe("Selecting/removing a team", () => {
    let component;

    beforeEach(() => {
        component = mount(
            <CollectionCreateController {...defaultProps} />
        );
        component.setProps({allTeams: mockedTeams});
    });
    
    it("disables a team on selection", () => {
        const selectedTeamID = "2";
        const getDisabledTeams = () => component.state('updatedAllTeams').filter(team => team.disabled);

        // Check that the component has the state and props we'd expect at this point
        expect(component.prop('allTeams').length).toBe(2);
        expect(component.state('updatedAllTeams')).toBe(null);

        component.instance().handleTeamSelection({preventDefault: ()=>{}, target: {value: selectedTeamID}});

        expect(component.state('updatedAllTeams')).toBeTruthy();
        expect(getDisabledTeams().length).toBe(1);
        expect(getDisabledTeams()[0].id).toBe("2");
        expect(getDisabledTeams()[0].name).toBe("Team 2");
    });

    it("enables a team on removal", () => {
        const removedTeam = {id: "2", name: "Team 2"};
        const getDisabledTeams = () => component.state('updatedAllTeams').filter(team => team.disabled);

        // Ensure that the component has the state and props we'd expect at this point
        expect(component.state('updatedAllTeams')).toBe(null);
        component.instance().handleTeamSelection({preventDefault: ()=>{}, target: {value: removedTeam.id}});
        expect(getDisabledTeams().length).toBe(1);
        expect(component.state('updatedAllTeams')).toBeTruthy();

        component.instance().handleRemoveTeam(removedTeam);
        expect(getDisabledTeams().length).toBe(0);
    })
});

test("Handle team selection doesn't add a team that is already in the collection", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    component.setProps({ allTeams: mockedTeams });
    const teams = [{id: "1", name: "Team 1"}];
    const newCollection = {...newCollectionDetails, teams: teams};
    component.setState({ newCollectionDetails: newCollection });
    expect(component.update().state().newCollectionDetails.teams).toHaveLength(1);
    component.instance().handleTeamSelection({preventDefault: ()=>{}, target: {value: "1"}});
    expect(component.update().state().newCollectionDetails.teams).toHaveLength(1);
});

test("Handle schedule type change updates state correctly", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    expect(component.update().state().newCollectionDetails.scheduleType).toBe("custom-schedule");
    component.instance().handleScheduleTypeChange({value: "calender-entry-schedule"});
    expect(component.update().state().newCollectionDetails.scheduleType).toBe("calender-entry-schedule");
});

test("Handle publish date change updates state correctly", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    expect(component.update().state().newCollectionDetails.publishDate.value).toBe("");
    component.instance().handlePublishDateChange({target: {value: "2017-12-07"}});
    expect(component.update().state().newCollectionDetails.publishDate.value).toBe("2017-12-07");
});

test("Handle publish time change updates state correctly", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    expect(component.update().state().newCollectionDetails.publishTime.value).toBe("09:30");
    component.instance().handlePublishTimeChange({target: {value: "13:00"}});
    expect(component.update().state().newCollectionDetails.publishTime.value).toBe("13:00");
});

test("Handle submit validates that a collection has a name", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    component.instance().handleSubmit({preventDefault: ()=>{}});
    expect(component.update().state().newCollectionDetails.name.errorMsg).toBe("Collections must be given a name");
});

test("Handle submit validates that a scheduled collection has a date", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    const name = {value: "Test Collection", errorMsg: ""};
    const newCollection = {...newCollectionDetails, name: name};
    component.setState({ newCollectionDetails: newCollection});
    component.instance().handleSubmit({preventDefault: ()=>{}});
    expect(component.update().state().newCollectionDetails.publishDate.errorMsg).toBe("Scheduled collections must be given a publish date");
});

test("Handle submit validates that a scheduled collection has a time", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    const name = {value: "Test Collection", errorMsg: ""};
    const date = {value: "2017-12-07", errorMsg: ""};
    const time = {value: "", errorMsg: ""};
    const newCollection = {...newCollectionDetails, name: name, publishDate: date, publishTime: time};
    component.setState({ newCollectionDetails: newCollection});
    component.instance().handleSubmit({preventDefault: ()=>{}});
    expect(component.update().state().newCollectionDetails.publishTime.errorMsg).toBe("Scheduled collections must be given a publish time");
});

test("Make publish date returns correct date value", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    const date = {value: "2017-12-07", errorMsg: ""};
    const time = {value: "09:30", errorMsg: ""};
    const newCollection = {...newCollectionDetails, publishDate: date, publishTime: time};
    component.setState({ newCollectionDetails: newCollection});
    const makePublishDate = component.instance().makePublishDate();
    expect(makePublishDate).toEqual("2017-12-07T09:30:00.000Z");
});

test("Map state to post body returns correct object", () => {
    const component = shallow(
        <CollectionCreateController {...defaultProps} />
    );
    const name = {value: "Test collection", errorMsg: ""};
    const date = {value: "2017-12-07", errorMsg: ""};
    const time = {value: "09:30", errorMsg: ""};
    const teams = [{id: "1", name: "Team 1"}];
    const newCollection = {...newCollectionDetails, name: name, publishDate: date, publishTime: time, teams: teams};
    component.setState({ newCollectionDetails: newCollection});
    const makePublishDate = component.instance().mapStateToPostBody();
    expect(makePublishDate).toEqual(expectedPostBody);
});