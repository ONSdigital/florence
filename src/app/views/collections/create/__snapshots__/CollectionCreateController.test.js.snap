// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create collection form matches stored snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="collection-name"
    >
      Collection name
    </label>
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input"
      disabled={false}
      id="collection-name"
      name="collection-name"
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="collection-teams"
    >
      Select a team(s) that can view this collection
    </label>
    <div
      className="select-wrap "
    >
      <select
        className="select"
        disabled={false}
        id="collection-teams"
        name="collection-teams"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="default-option"
      >
        <option
          value="Loading teams..."
        >
          Loading teams...
        </option>
      </select>
    </div>
  </div>
  <div
    className="selected-item-list"
  />
  <fieldset
    className="fieldset"
  >
    <legend
      className="fieldset__legend"
    >
      Publish type
    </legend>
    <div
      className="radio radio--inline"
    >
      <input
        checked={true}
        className="radio__input selected"
        id="scheduled-radio"
        name="release-type"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="scheduled"
      />
      <label
        className="radio__label radio__label--selected"
        htmlFor="scheduled-radio"
      >
        Scheduled publish
         
      </label>
    </div>
    <div
      className="radio radio--inline"
    >
      <input
        checked={false}
        className="radio__input"
        id="manual-radio"
        name="release-type"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="manual"
      />
      <label
        className="radio__label"
        htmlFor="manual-radio"
      >
        Manual publish
         
      </label>
    </div>
  </fieldset>
  <div>
    <fieldset
      className="fieldset"
    >
      <legend
        className="fieldset__legend"
      >
        Schedule type
      </legend>
      <div
        className="radio radio--inline"
      >
        <input
          checked={true}
          className="radio__input selected"
          id="custom-radio"
          name="schedule-type"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="custom-schedule"
        />
        <label
          className="radio__label radio__label--selected"
          htmlFor="custom-radio"
        >
          Custom schedule
           
        </label>
      </div>
      <div
        className="radio radio--inline"
      >
        <input
          checked={false}
          className="radio__input"
          id="calendar-radio"
          name="schedule-type"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="calender-entry-schedule"
        />
        <label
          className="radio__label"
          htmlFor="calendar-radio"
        >
          Calendar entry schedule
           
        </label>
      </div>
    </fieldset>
    <div>
      <div
        className="form__input"
      >
        <label
          className="form__label"
          htmlFor="publish-date"
        >
          Publish date
        </label>
        
        <input
          autoComplete="new-password"
          autoFocus={false}
          className="input"
          disabled={false}
          id="publish-date"
          max="2017-10-12"
          min="2017-10-12"
          name="publish-date"
          onChange={[Function]}
          onFocus={[Function]}
          type="date"
          value=""
        />
        
      </div>
      <div
        className="form__input"
      >
        <label
          className="form__label"
          htmlFor="publish-time"
        >
          Publish time
        </label>
        
        <input
          autoComplete="new-password"
          autoFocus={false}
          className="input"
          disabled={false}
          id="publish-time"
          name="publish-time"
          onChange={[Function]}
          onFocus={[Function]}
          type="time"
          value="09:30"
        />
        
      </div>
    </div>
  </div>
  <button
    className="btn btn--positive margin-top--1"
    disabled={false}
    type="submit"
  >
    Create collection
  </button>
  
</form>
`;
